shared/utils/checkStatus.ts

```diff
@@ -0,0 +1,33 @@
+import { capitalize } from "./capitalize";
+import { Types } from "@requestnetwork/request-client.js";
+
+export const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+  const balance = Number(request?.balance?.balance ?? 0);
+  const expectedAmount = Number(request?.expectedAmount ?? 0);
+
+  if (balance > 0) {
+    return balance >= expectedAmount ? "Paid" : "Partially Paid";
+  }
+
+  const eventStatus = {
+    reject: "Rejected",
+    cancel: "Canceled",
+  };
+
+  const dueDate = request?.contentData?.paymentTerms?.dueDate;
+  if (dueDate && new Date(dueDate) < new Date()) {
+    return "Overdue";
+  }
+
+  for (const [event, status] of Object.entries(eventStatus)) {
+    if (
+      request?.events?.some(
+        (e: { name?: string }) => e?.name?.toLowerCase() === event.toLowerCase()
+      )
+    ) {
+      return status;
+    }
+  }
```

Comment:
No need for this

Reply:
I actually kind of like the `lastEvent` variable - it seems easier to understand. It might work well with the pseudocode I wrote in my previous comment. But I leave the final decision on implementation up to you.